### PR TITLE
feat: Workspace stats and devstats skeleton

### PR DESCRIPTION
**Pull request by jpmcb**
## Description

This PR implements workspaces stats and bootstraps the skeleton for devstats health metrics.

Workspace stats:
- Adds the `DbIssuesGitHubEvents` and `DbPushGitHubEvents` entities
- Adds a `findIssueStatsByRepo` for the issues event service
- Adds the workspace stats endpoint, controller, and service

Devstats:
- Adds `mathjs` for doing calculations (primarily only needed right now for the `quantileSeq` function)
- Adds a sample set of 1000 top repos and their activity ratio score
- Adds winsorizing of activity ratio data from the sample set and normalizes it to 0-10 
- `v2/repos/search` now has this winsorized/normalized `activity_ratio` 

Misc:
- Renames "IssueGithubEvents" to be "IssuesGithubEvents" which mirrors the database table name (note the trailing `s` in "issues")

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes: https://github.com/open-sauced/api/issues/520
Related to: https://github.com/open-sauced/app/pull/2546

## Mobile & Desktop Screenshots/Recordings

Screenshot 2024-01-30 at 9 49 21 AM

## Steps to QA

After authenticating and getting the ID to a workspace, hit the workspaces stats endpoint:

```
curl -X 'GET' \
  'http://localhost:3003/v2/workspaces/cbf33074-0016-4a05-8fe3-4f329a2ef6ae/stats?range=30&prev_days_start_date=0' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer <redacted>'
```

For this workspace, this has a few low activity repos and 1 very high activity repo (kubernetes/kubernetes):

```json
{
  "pull_requests": {
    "opened": 145,
    "merged": 263,
    "velocity": 10.75
  },
  "issues": {
    "opened": 108,
    "closed": 199,
    "velocity": 83.5
  },
  "repos": {
    "stars": 105470,
    "forks": 39331,
    "health": 4.5781695677167615
  }
}
```

Repo search stats also include the activity ratio now too:

```
curl -X 'GET' \
  'http://localhost:3003/v2/repos/search?page=1&limit=10&range=30&prev_days_start_date=0&repo=kubernetes%2Fkubernetes' \
  -H 'accept: application/json'
``` 

```
      "pr_active_count": 153,
      "open_prs_count": 134,
      "merged_prs_count": 172,
      "spam_prs_count": 0,
      "draft_prs_count": 19,
      "closed_prs_count": 89,
      "pr_velocity_count": 36,
      "activity_ratio": 10
```

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/timescale/repo-devstats.service.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+    // calculate quantile values using the constant sampled activity ratios
+    this.lowerBound = math.quantileSeq(avgRepoActivityRatioSample, lowerQ) as number;
+    this.upperBound = math.quantileSeq(avgRepoActivityRatioSample, upperQ) as number;
+  }
+
+  private lowerBound: number;
+  private upperBound: number;
+
+  private winsorizeAndNormalizeRatio(ratio: number): number {
```
Comment (nickytonline) [inline src/timescale/repo-devstats.service.ts:32]: TIL about [winsorizing](https://en.wikipedia.org/wiki/Winsorizing).

**Review by nickytonline — APPROVED** [review]
I think this looks good. I gave a read through. Let's get it into workspaces and see the magic.

File: src/timescale/repo-devstats.service.ts (hunk 1 above)
Comment (jpmcb) [reply src/timescale/repo-devstats.service.ts:32]: Shout out to @bdougie for making this statistical magic happen 👀

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**